spec/components/composer_messages_finder_spec.rb

```diff
@@ -467,4 +467,20 @@
     end
   end
 
+  context 'when editing a post' do
+    let(:user) { Fabricate.build(:user) }
+    let(:new_post_finder) { ComposerMessagesFinder.new(user, composer_action: 'reply') }
+    let(:edit_post_finder) { ComposerMessagesFinder.new(user, composer_action: 'edit') }
+
+    before do
+      SiteSetting.educate_until_posts = 10
+    end
+
+    it "returns nothing even if it normally would" do
+      user.expects(:post_count).returns(9)
```

Comment:
We prefer to avoid stubs in our code base. What I think we can do here is to create a user with 2 posts and reduce the setting for `SiteSetting.educate_until_posts`